src/Uno.Toolkit.RuntimeTests/Tests/ItemsRepeaterChipTests.cs

```diff
@@ -0,0 +1,177 @@
+		Assert.IsTrue(actual.All(x => x == false));
+
+		((ChipControl)SUT.TryGetElement(1)).Toggle();
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+	}
+
+	#endregion
+
+	#region Changing SelectionMode
+
+	[TestMethod]
+	public async Task SingleOrNoneToSingle_NoneSelected_ShouldAutoSelect()
+	{
+		var source = Enumerable.Range(0, 3).ToArray();
+		var SUT = SetupItemsRepeater(source, ChipSelectionMode.SingleOrNone);
+		bool?[] expected = new bool?[] { true, false, false }, actual;
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+		actual = GetChipsSelectionState(SUT);
+		Assert.IsTrue(actual.All(x => x == false));
+
+		ChipExtensions.SetChipSelectionMode(SUT, ChipSelectionMode.Single);
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+	}
+
+	[TestMethod]
+	public async Task SingleOrNoneToSingle_Selected_ShouldPreserveSelection()
+	{
+		var source = Enumerable.Range(0, 3).ToArray();
+		var SUT = SetupItemsRepeater(source, ChipSelectionMode.SingleOrNone);
+		bool?[] expected = new bool?[] { false, false, true }, actual;
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+		((ChipControl)SUT.TryGetElement(2)).Toggle();
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+
+		ChipExtensions.SetChipSelectionMode(SUT, ChipSelectionMode.Single);
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+	}
+
+	[TestMethod]
+	public async Task MultiToSingle_Selected_ShouldPreserveFirstSelection()
+	{
+		var source = Enumerable.Range(0, 3).ToArray();
+		var SUT = SetupItemsRepeater(source, ChipSelectionMode.Multiple);
+		bool?[] expected = new bool?[] { false, true, true }, actual;
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+		((ChipControl)SUT.TryGetElement(1)).Toggle();
+		((ChipControl)SUT.TryGetElement(2)).Toggle();
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+
+		ChipExtensions.SetChipSelectionMode(SUT, ChipSelectionMode.Single);
+		expected = new bool?[] { false, true, false };
+		actual = GetChipsSelectionState(SUT);
+		CollectionAssert.AreEqual(expected, actual);
+	}
+
+	#endregion
+
+	private static ItemsRepeater SetupItemsRepeater(object source, ChipSelectionMode mode)
+	{
+		var SUT = new ItemsRepeater
+		{
+			ItemsSource = source,
+			ItemTemplate = XamlHelper.LoadXaml<DataTemplate>(@"
+				<DataTemplate>
+					<Chip xmlns='using:Uno.Toolkit.UI' />
+				</DataTemplate>
+			"),
+		};
+		ChipExtensions.SetChipSelectionMode(SUT, mode);
+
+		return SUT;
+	}
+
+	private static bool? IsChipSelectedAt(ItemsRepeater ir, int index)
+	{
+		return (ir.TryGetElement(index) as ChipControl)?.IsChecked;
+	}
+
+	// since we are not using IsThreeState=True, the values can only be true/false.
+	// if any of them is null, that means there other problem and should be thrown.
+	// therefore, only == check should be used in assertion.
```

Comment:
```suggestion
	// since we are not using IsThreeState=True, the values can only be true/false.
	// if any of them is null, that means there is another problem and should be thrown.
	// therefore, only == check should be used in an assertion.
```